daemon/graphdriver/driver.go

```diff
@@ -112,6 +112,21 @@ type Driver interface {
 	DiffDriver
 }
 
+// Capabilities defines a list of capabilities a driver may implement.
+// These capabilities are not required; however, they do determine how a
+// graphdriver can be used.
+type Capabilities struct {
+	// Flags that this driver is capable of reproducing exactly equivalent
+	// diffs for read-only layers.
+	ReproducesExactDiffs bool
```

Comment:
Maybe add an extra doc comment like 

```
In practice when true, relies on the graphdriver to provide consistent tar streams for a layer rather than the generic algorithm the layer store provides
```

I don't usually like stuff like this that refers to "parent" packages, but is helpful for readers at least.